src/blueapi/client/numtracker.py

```diff
@@ -0,0 +1,79 @@
+import logging
+from collections.abc import Mapping
+from pathlib import Path
+from textwrap import dedent
+
+import requests
+from pydantic import Field
+
+from blueapi.utils import BlueapiBaseModel
+
+
+class NumtrackerNewScanVisit(BlueapiBaseModel):
+    instrument: str
+    instrument_session: str = Field(alias="instrumentSession")
+    path: Path
+
+
+class NumtrackerNewScanScan(BlueapiBaseModel):
+    scan_file: str = Field(alias="scanFile")
+    scan_number: int = Field(alias="scanNumber")
+    directory: NumtrackerNewScanVisit
+
+
+class NumtrackerNewScan(BlueapiBaseModel):
+    scan: NumtrackerNewScanScan
```

Comment:
Could: Rename these to align with [the numtracker schema](https://github.com/DiamondLightSource/numtracker/blob/main/static/service_schema) and the wrapping one as a generic response class:

```suggestion
class VisitPath(BlueapiBaseModel):
    instrument: str
    instrument_session: str = Field(alias="instrumentSession")
    path: Path


class ScanPaths(BlueapiBaseModel):
    scan_file: str = Field(alias="scanFile")
    scan_number: int = Field(alias="scanNumber")
    directory: VisitPath


class NumtrackerScanMutationResponse(BlueapiBaseModel):
    scan: ScanPaths
```

Reply:
These have now changed names in the numtracker API, and I'd support moving to those names 

VisitPath -> DirectoryPath